pkg/encryption/app.go

```diff
@@ -0,0 +1,136 @@
+package encryption
+
+import (
+	"context"
+	"fmt"
+
+	chartv1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/k8smetadata/pkg/annotation"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/encryption-provider-operator/pkg/project"
+)
+
+const (
+	configMapName  = "encryption-config-hasher-chart-values"
+	chartNamespace = "giantswarm"
+
+	appName      = "encryption-config-hasher"
+	appNamespace = "kube-system"
+
+	encryptionConfigHasherVersion = "0.1.1"
+)
+
+var (
+	chartURL = fmt.Sprintf("https://giantswarm.github.io/giantswarm-playground-catalog/encryption-config-hasher-%s.tgz", encryptionConfigHasherVersion)
+)
+
+func (s *Service) deployEncryptionProviderHasherApp(ctx context.Context, wcClient ctrlclient.Client) error {
+	cm := buildConfigMapValues(s.registryDomain)
+
+	err := wcClient.Create(ctx, cm)
+	if apierrors.IsAlreadyExists(err) {
+		// fall through
+	} else if err != nil {
+		return err
+	}
+
+	chart := buildAppChart()
+
+	err = wcClient.Create(ctx, chart)
+
+	if apierrors.IsAlreadyExists(err) {
```

Comment:
How will you handle upgrades of the app?

Reply:
the app is deleted after the migration is finished so it does not run there for no reason

Reply:
do you see any problems/issues with that?

Reply:
OK make sense